### PR TITLE
UI Changes and New Search Bar

### PR DESCRIPTION
**Pull request by katy-frank**
Basic UI changes (the display cards, edits to the dataset and sample pages) and a new search bar for the homepage that enables searching datasets and samples using different terms (water bodies, sample type, location, name, reference, etc).

### CODE REVIEW

File: mibios/glamr/forms.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 from django.core.exceptions import ValidationError
 from django.forms.widgets import Select, TextInput
 # from django.utils.safestring import mark_safe
-
+from django.forms import ModelForm
+from django.db.models import Q
```
Comment (robert102) [inline mibios/glamr/forms.py:6]: empty lines around django import block, within block import order is alphabetical
```suggestion
from django.db.models import Q
from django.forms import ModelForm

```

File: mibios/glamr/views.py [hunk 2]
```diff
@@ -906,19 +905,106 @@ def search(self, check_abundance=False):
...
+                hits = [
+                    (i.content_object, str(i.content_object), None)
+                    for i in grp
+                ]
+                self.hits.append((
+                    have_abundance,
+                    model._meta.verbose_name_plural,
+                    model._meta.model_name,
+                    hits,
+                ))
```
Comment (robert102) [inline mibios/glamr/views.py:920]: don't change the indentation, this is a for ... else block, the else does not belong to the if, the else block is skipped if the for loop is exited with a break.

File: mibios/glamr/templates/glamr/sample_detail.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{% extends "glamr/base.html" %}
+{% block content %}
+
+<nav aria-label="breadcrumb">
+  <ol class="breadcrumb">
+    <li class="breadcrumb-item"><a href="/">Home</a></li>
+    <li class="breadcrumb-item"><a href="/{{ detail_url }}">Data Set</a></li>
```
Comment (robert102) [inline mibios/glamr/templates/glamr/sample_detail.html:7]: I did quite follow the changes to the SampleView class, does your get_detail_url() do anything that the url tag suggested below doesn't?
```suggestion
    <li class="breadcrumb-item"><a href="{% url 'dataset' pk=object.dataset.pk %}">Data Set</a></li>
```

File: mibios/glamr/templates/glamr/sample_detail.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    <li class="breadcrumb-item"><a href="/{{ detail_url }}samples">Samples</a></li>
```
Comment (robert102) [inline mibios/glamr/templates/glamr/sample_detail.html:8]:
```suggestion
    <li class="breadcrumb-item"><a href="{% url 'dataset_sample_list' pk=object.dataset.pk %}">Samples</a></li>
```

File: mibios/glamr/models.py [hunk 5]
```diff
@@ -66,7 +66,7 @@ class Dataset(AbstractDataset):
     note = models.TextField(**ch_opt)
 
     loader = DatasetLoader()
-    orphan_group_description = 'samples without a data set'
+    orphan_group_description = 'Samples without a data set'
```
[the remark this answers is not in the document]
Comment (katy-frank) [reply mibios/glamr/models.py:75]: Fixed; capfirst doesn't work when the field is linkified (as description is, to link to dataset), so I am now using the render_scheme function in tables.py to handle this appropriately

File: mibios/glamr/tables.py [hunk 6]
```diff
@@ -154,16 +178,24 @@ class Meta:
 
     def render_scheme(self, value, record):
         r = record
-        return r.scheme or r.short_name or r.bioproject \
+        scheme = r.scheme or r.short_name or r.bioproject \
             or r.jgi_project or r.gold_id or str(record)
+        return string.capwords(scheme)
```
[the remark this answers is not in the document]
Comment (katy-frank) [reply mibios/glamr/tables.py:191]: I'm of the opinion that ensuring proper capitalization here is a good idea, unless we have plans to go through and ensure consistent capitalization in the spreadsheet. I can remove this if you feel strongly about it, but I was noticing inconsistent capitalization without it

File: mibios/glamr/templates/glamr/sample_detail.html (hunk 3 above)
Comment (katy-frank) [reply mibios/glamr/templates/glamr/sample_detail.html]: It does not- thanks for the suggestion, I didn't know how to do this! I've removed get_detail_url and now do this instead

File: mibios/glamr/tables.py [hunk 7]
```diff
@@ -12,7 +12,7 @@ def get_record_url(*args):
     Return URL for an object
 
     Arguments: <obj> | <<model|model_name> <pk>>
-
+no dataset / study information available
```
[the remark this answers is not in the document]
Comment (katy-frank) [reply mibios/glamr/tables.py]: Yes, good catch! Removed

File: mibios/glamr/templates/glamr/table_cards.html [hunk 8]
```diff
@@ -42,11 +42,11 @@
                             {% for column in table.columns %}
                                     {% if column.orderable %}
                                         <li class="nav-item">
-                                            <a class="sort nav-link" id={{ column.attrs.navID }} data-toggle="pill" href="{% querystring table.prefixed_order_by_field=column.order_by_alias.next %}">{{ column.header }}<i id="{{column.attrs.navID}}-icon" class="icon-sort bi bi-caret-down m-1"></i></a>
```
[the remark this answers is not in the document]
Comment (katy-frank) [reply mibios/glamr/templates/glamr/table_cards.html]: You're right about this- I think this commit message is inaccurate. I believe that at the time, running either git status or my editing environment led me to believe I had not actually committed this file yet at all, when I definitely had.

File: mibios/glamr/views.py [hunk 9]
```diff
@@ -494,14 +495,84 @@ def get_details(self):
...
+                    detail[2]
+                ))
+            if detail[0] == 'note' and detail[2]:
+                ordered_details.append((
+                    'Notes',
+                    detail[1],
+                    detail[2]
+                ))
+
+        return ordered_details, rel_lists
```
[the remark this answers is not in the document]
Comment (katy-frank) [reply mibios/glamr/views.py:581]: Yes I can definitely implement this a better way! Which class would I be adding this field_order attribute to?

File: mibios/glamr/views.py [hunk 10]
```diff
@@ -493,14 +501,86 @@ def get_details(self):
...
+                ordered_details.append((
+                    'Gold ID',
+                    detail[1],
+                    detail[2]
+                ))
+            if detail[0] == 'material type' and detail[2]:
+                value_text = detail[2]
+                if value_text.lower() == 'edna':
+                    value_text = "eDNA"
+                else:
```
Comment (robert102) [inline mibios/glamr/views.py:539]: for those things (there are others) I'll add a template tag, "capfirstkeep" that will work like capfirst except if the value contains an upper case letter already, then it keeps the exoisting value.

File: mibios/glamr/views.py [hunk 11]
```diff
@@ -493,14 +501,86 @@ def get_details(self):
...
-            'Samples',
-            reverse('dataset_sample_list', args=[pk]),
-            f'List of {self.object.samples().count()} samples'
-        )] + details
-        return details, rel_lists
 
+        # this is hacky but want details in particular order
+        ordered_details = []
+        # reference, samples, bioproject, gold id, material type, water bodies,
+        # primers, sequencing target, sequencing platform, size fraction, notes
```
Comment (robert102) [inline mibios/glamr/views.py:508]: I've just commited to main branch a field ordering for the detail branch, so just set field_order = [...]

**Review by robert102 — CHANGES_REQUESTED** [review]

**Pull request merged**